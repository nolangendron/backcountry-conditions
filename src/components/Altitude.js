import React from "react";
import styled from "@emotion/styled";

const AltitudeStyles = styled("div")``;

const Altitude = () => {
  return <AltitudeStyles></AltitudeStyles>;
};

export default Altitude;
