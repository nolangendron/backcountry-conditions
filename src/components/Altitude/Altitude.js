import React from "react";
import styled from "@emotion/styled";

const AltitudeStyles = styled("h5")`
  grid-column: 1 / span 1;
  grid-row: ${(props) => (props.elevationLower ? "3 / span 1" : "2 / span 1")};
  justify-self: center;
  align-self: center;
  font-weight: 300;
`;

const Altitude = ({ elevation, elevationLower }) => {
  return (
    <AltitudeStyles elevationLower={elevationLower}>{elevation}</AltitudeStyles>
  );
};

export default Altitude;
