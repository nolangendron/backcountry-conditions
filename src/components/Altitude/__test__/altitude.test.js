import React from "react";
import ReactDOM from "react-dom";
import Altitude from "../Altitude.js";

import { render, cleanup } from "@testing-library/react";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Altitude />, div);
});

it("renders Altitude correctly", () => {
  render(<Altitude elevation="undefined" />);
});

it("matches snapshot", () => {
  const tree = renderer
    .create(<Altitude elevation="1200" elevationLower="false" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
