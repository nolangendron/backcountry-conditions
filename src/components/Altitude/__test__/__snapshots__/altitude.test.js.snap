// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="css-u7c6m7"
>
  <h3
    className="elevation"
  >
    1200
  </h3>
</div>
`;
